webapp/channels/src/components/edit_category_modal/edit_category_modal.tsx

```diff
@@ -54,12 +58,35 @@ export default class EditCategoryModal extends React.PureComponent<Props, State>
         this.handleClear();
     };
 
+    handleEnterKeyEvent = (e?: React.KeyboardEvent<HTMLDivElement>) => {
+        if (e && e.nativeEvent.isComposing) {
+            return;
+        }
+        this.handleConfirm();
+    };
+
+    componentWillUnmount() {
+        if (this.timeoutId) {
+            clearTimeout(this.timeoutId);
+        }
+    }
+
     handleConfirm = () => {
+        if (this.state.isProcessing) {
```

Comment:
We don't need re-renders in case of changing to `isProcessing`, so... wouldn't it be better to make this just an object property (similar to the timeoutId), instead of a state? (what would be in functional components a ref)

Reply:
Yeah, I'd agree with just making it a regular member variable. I think there's also a risk with state that `this.state` wouldn't always be updated by the time that the second call to `handleConfirm` happens with how React sometimes batches state updates

Reply:
of course, thanks for the feedback!